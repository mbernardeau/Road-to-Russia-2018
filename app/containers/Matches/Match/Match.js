import React, {
  Component,
} from 'react';

import PropTypes from 'prop-types';
import _ from 'lodash';

import Card, {
  CardContent,
} from 'material-ui/Card';

import Divider from 'material-ui/Divider';

import Bet from './Bet';
import ValidIcon from './ValidIcon';
import MatchInfos from './MatchInfos';

const empty = {};

class Match extends Component {
  constructor(props) {
    super(props);

    this.state = {
      bet: empty,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (!this.state.bet || this.state.bet === empty) {
      this.setState({
        bet: nextProps.bet || empty,
      });
    }
  }

  shouldComponentUpdate(nextProps, nextState) {
    return this.state.bet.teamA !== nextState.bet.teamA || this.state.bet.teamB !== nextState.bet.teamB;
  }

  isBetValid = () => {
    const scoreValidator = (score) => _.isNumber(score) && score >= 0;

    return _.conformsTo(this.state.bet, {
      teamA: scoreValidator,
      teamB: scoreValidator,
    });
  }

  handleChange = (team) => ({ target: { value } }) => {
    this.setState({
      bet: {
        ...this.state.bet,
        [`team${team}`]: value,
      },
    }, this.saveBetIfValid);
  };

  saveBetIfValid = () => {
    if (this.isBetValid()) {
      this.props.saveBet(this.state.bet);
    }
  }

  handleTeamAChange = this.handleChange('A');
  handleTeamBChange = this.handleChange('B');

  render() {
    const { match, matchId } = this.props;
    const { bet } = this.state;

    return (
      <Card style={styles.card}>
        <CardContent style={styles.cardContainer}>
          <div style={styles.match}>
            <Bet team={match.teamA} betValue={bet.teamA} onBetValueUpdated={this.handleTeamAChange} />
            <Bet team={match.teamB} betValue={bet.teamB} onBetValueUpdated={this.handleTeamBChange} />
          </div>
          <Divider />
          <MatchInfos match={match} matchId={matchId} />
          <ValidIcon valid={this.isBetValid()} />
        </CardContent>
      </Card>
    );
  }
}

const styles = {
  match: {
    display: 'grid',
    gridTemplateColumns: '50% 50%',
    marginBottom: 10,
  },

  card: {
    marginTop: 7,
    marginBottom: 7,
    width: '100%',
  },

  cardContainer: {
    position: 'relative',
  },
};

Match.propTypes = {
  match: PropTypes.shape({
    teamA: PropTypes.shape({
      code: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
    teamB: PropTypes.shape({
      code: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
  }),
  bet: PropTypes.shape({
    teamA: PropTypes.number,
    teamB: PropTypes.number,
  }),
  saveBet: PropTypes.func.isRequired,
  matchId: PropTypes.string,
};

Match.defaultProps = {
  bet: empty,
};


export default Match;
