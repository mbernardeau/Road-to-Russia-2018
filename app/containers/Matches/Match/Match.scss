.match {
  display: grid;
  grid-template-columns: 50% 50%;
  height: 100%;
}
