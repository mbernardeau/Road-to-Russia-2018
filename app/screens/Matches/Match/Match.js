import React, { Component } from 'react'

import PropTypes from 'prop-types'

import { conformsTo, isNumber } from 'lodash'

import Card, { CardContent } from 'material-ui/Card'

import Divider from 'material-ui/Divider'

import Bet from './Bet'
import ValidIcon from './ValidIcon'
import MatchInfos from './MatchInfos'

const empty = {}

class Match extends Component {
  constructor(props) {
    super(props)

    this.state = {
      bet: empty,
    }
  }

  componentWillReceiveProps(nextProps) {
    if (!this.state.bet || this.state.bet === empty) {
      this.setState({
        bet: nextProps.bet || empty,
      })
    }
  }

  isBetValid = () => {
    const scoreValidator = score => isNumber(score) && score >= 0

    return conformsTo(this.state.bet, {
      teamA: scoreValidator,
      teamB: scoreValidator,
    })
  }

  handleChange = team => ({ target: { value } }) => {
    this.setState(
      {
        bet: {
          ...this.state.bet,
          [`team${team}`]: value,
        },
      },
      this.saveBetIfValid,
    )
  }

  saveBetIfValid = () => {
    if (this.isBetValid()) {
      this.props.saveBet(this.state.bet)
    }
  }

  betSaved = () =>
    this.isBetValid() &&
    this.state.bet.teamA === this.props.bet.teamA &&
    this.state.bet.teamB === this.props.bet.teamB

  handleTeamAChange = this.handleChange('A')
  handleTeamBChange = this.handleChange('B')

  render() {
    const { match, teamA, teamB } = this.props
    const { bet } = this.state

    return (
      <Card style={styles.card}>
        <CardContent style={styles.cardContainer}>
          <div style={styles.match}>
            <Bet team={teamA} betValue={bet.teamA} onBetValueUpdated={this.handleTeamAChange} />
            <Bet team={teamB} betValue={bet.teamB} onBetValueUpdated={this.handleTeamBChange} />
          </div>
          <Divider />
          <MatchInfos match={match} />
          <ValidIcon valid={this.betSaved()} />
        </CardContent>
      </Card>
    )
  }
}

const styles = {
  match: {
    display: 'grid',
    gridTemplateColumns: '50% 50%',
    marginBottom: 10,
  },

  card: {
    marginTop: 7,
    marginBottom: 7,
    width: '100%',
  },

  cardContainer: {
    position: 'relative',
  },
}

Match.defaultProps = {
  match: {},
  teamA: {},
  teamB: {},
}

Match.propTypes = {
  match: PropTypes.object,
  teamA: PropTypes.shape({
    code: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }),
  teamB: PropTypes.shape({
    code: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }),
  bet: PropTypes.shape({
    teamA: PropTypes.number,
    teamB: PropTypes.number,
  }),
  saveBet: PropTypes.func.isRequired,
}

Match.defaultProps = {
  bet: empty,
}

export default Match
