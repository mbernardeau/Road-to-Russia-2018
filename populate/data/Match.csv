Index;Phase;PaysA;PaysB;ScoreA;ScoreB;Winner;Cote1;CoteN;Cote2;CoteA;CoteB;Dateheure;Stade
1;0;1;2;-1;-1;0;1.25;4.8;9;0;0;2018-06-14 16:00:00;Stade Loujniki
2;0;3;4;-1;-1;0;2;3;3.5;0;0;2018-06-15 13:00:00;Iekaterinbourg Arena
3;0;5;6;-1;-1;0;1.85;3.1;3.9;0;0;2018-06-15 16:00:00;Stade Krestovski
4;0;7;8;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-15 19:00:00;Stade olympique Ficht
5;0;13;14;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-16 11:00:00;Kazan-Arena
6;0;11;12;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-16 17:00:00;Stade Yubileyniy
7;0;9;10;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-16 14:00:00;Otkrytie Arena
8;0;15;16;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-16 20:00:00;Stade de Kaliningrad
9;0;17;18;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-17 19:00:00;Stade de Rostov-sur-le-Don
10;0;19;20;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-17 13:00:00;Stade de Samara
11;0;21;22;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-17 16:00:00;Stade Loujniki
12;0;23;24;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-18 13:00:00;Stade de Nijni-Novgorod
13;0;8;6;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-18 16:00:00;Stade olympique Ficht
14;0;2;4;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-18 19:00:00;Stade central
15;0;1;3;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-19 13:00:00;Otkrytie Arena
16;0;7;5;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-19 16:00:00;Stade Yubileyniy
17;0;10;12;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-19 19:00:00;Stade Krestovski
18;0;9;11;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-20 16:00:00;Stade de Rostov-sur-le-Don
19;0;20;18;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-20 13:00:00;Stade Loujniki
20;0;16;14;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-20 19:00:00;Kazan-Arena
21;0;15;13;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-21 13:00:00;Iekaterinbourg Arena
22;0;19;17;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-21 16:00:00;Stade de Samara
23;0;24;22;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-21 19:00:00;Stade de Nijni-Novgorod
24;0;23;21;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-22 16:00:00;Stade central
25;0;2;3;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-22 13:00:00;Stade Krestovski
26;0;4;1;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-22 19:00:00;Stade de Kaliningrad
27;0;8;5;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-23 16:00:00;Stade olympique Ficht
28;0;6;7;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-23 19:00:00;Stade de Rostov-sur-le-Don
29;0;10;11;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-23 13:00:00;Otkrytie Arena
30;0;12;9;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-24 13:00:00;Stade de Nijni-Novgorod
31;0;16;13;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-24 19:00:00;Kazan-Arena
32;0;14;15;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-24 16:00:00;Iekaterinbourg Arena
33;0;24;21;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-25 15:00:00;Stade de Samara
34;0;22;23;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-25 15:00:00;Stade central
35;0;20;17;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-25 19:00:00;Stade Yubileyniy
36;0;18;19;-1;-1;0;1.7;3.2;4.3;0;0;2018-06-25 19:00:00;Stade de Kaliningrad
37;0;16;13;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-26 15:00:00;Stade Loujniki
38;0;14;15;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-26 15:00:00;Stade olympique Ficht
39;0;12;9;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-26 19:00:00;Stade Krestovski
40;0;10;11;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-26 19:00:00;Stade de Rostov-sur-le-Don
41;0;24;21;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-27 19:00:00;Otkrytie Arena
42;0;22;23;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-27 19:00:00;Stade de Nijni-Novgorod
43;0;20;17;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-27 15:00:00;Kazan-Arena
44;0;18;19;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-27 15:00:00;Iekaterinbourg Arena
45;0;28;25;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-28 19:00:00;Stade de Kaliningrad
46;0;26;27;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-28 19:00:00;Stade Yubileyniy
47;0;32;29;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-28 15:00:00;Stade central
48;0;30;31;-1;-1;0;1.7;3.2;4.3;3.2;0;2018-06-28 15:00:00;Stade de Samara
49;8;22;3;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-06-30 19:00:00;Stade olympique Ficht
50;8;20;5;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-06-30 15:00:00;Kazan-Arena
51;8;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-01 15:00:00;Stade Loujniki
52;8;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-01 19:00:00;Stade de Nijni-Novgorod
53;8;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-02 15:00:00;Stade de Samara
54;8;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-02 19:00:00;Stade de Rostov-sur-le-Don
55;8;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-03 15:00:00;Stade Krestovski
56;8;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-03 19:00:00;Otkrytie Arena
57;4;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-06 15:00:00;Stade de Nijni-Novgorod
58;4;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-06 19:00:00;Kazan-Arena
59;4;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-07 19:00:00;Stade olympique Ficht
60;4;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-07 15:00:00;Stade de Samara
61;2;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-10 19:00:00;Stade Krestovski
62;2;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-11 19:00:00;Stade Loujniki
63;3;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-14 15:00:00;Stade Krestovski
64;1;15;1;-1;-1;0;1.7;3.2;4.3;3.2;4.3;2018-07-15 16:00:00;Stade Loujniki
