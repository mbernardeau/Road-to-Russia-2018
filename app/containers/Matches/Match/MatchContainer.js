import {
  isLoaded,
  firebaseConnect,
} from 'react-redux-firebase';

import { lazyload } from 'react-lazyload';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { get } from 'lodash';
import { toDotPath } from 'helpers/PathUtils';
import placeholder from 'components/Placeholder';

import Match from './Match';

const generateFirebasePath = ({ matchId, userId }) => `bets/${matchId}/users/${userId}`;

export default compose(
  lazyload({
    height: 150,
    once: true,
    offset: 300,
  }),
  connect(({ firebase: { auth: { uid } } }) => ({
    userId: uid,
  })),
  firebaseConnect(
    (props) => ({ path: generateFirebasePath(props) })
  ),
  connect(
    ({ firebase: { data } }, ownProps) => ({
      bet: get(data, toDotPath(generateFirebasePath(ownProps))),
    }),
    (dispatch, { firebase, ...props }) => ({
      saveBet: (newBet) => firebase.set(generateFirebasePath(props), newBet),
    }),
  ),
  placeholder({
    isLoaded: ({ match }) => isLoaded(match),
  })
)(Match);
