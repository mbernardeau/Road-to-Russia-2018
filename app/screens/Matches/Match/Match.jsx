import React, { Component } from 'react'

import PropTypes from 'prop-types'

import { conformsTo, isNumber } from 'lodash'

import Card from 'material-ui/Card'
import CardContent from 'material-ui/Card/CardContent'
import Divider from 'material-ui/Divider'
import moment from 'moment'

import Odds from './Odds'
import Bet from './Bet'
import ValidIcon from './ValidIcon'
import MatchInfos from './MatchInfos'
import Scores from './Scores'

import './Match.scss'

const empty = {}

class Match extends Component {
  constructor(props) {
    super(props)

    this.state = {
      bet: empty,
    }
  }

  componentWillReceiveProps(nextProps) {
    if (!this.state.bet || this.state.bet === empty) {
      this.setState({
        bet: nextProps.bet || empty,
      })
    }
  }

  isBetValid = () => {
    const scoreValidator = score => isNumber(score) && score >= 0

    return conformsTo(this.state.bet, {
      betTeamA: scoreValidator,
      betTeamB: scoreValidator,
    })
  }

  handleChange = team => ({ target: { value } }) => {
    this.setState(
      {
        bet: {
          ...this.state.bet,
          [`betTeam${team}`]: value,
        },
      },
      this.saveBetIfValid,
    )
  }

  saveBetIfValid = () => {
    if (this.isBetValid()) {
      this.props.saveBet(this.state.bet)
    }
  }

  betSaved = () =>
    this.isBetValid() &&
    this.state.bet.betTeamA === this.props.bet.betTeamA &&
    this.state.bet.betTeamB === this.props.bet.betTeamB

  handleTeamAChange = this.handleChange('A')
  handleTeamBChange = this.handleChange('B')

  render() {
    const { match, teamA, teamB } = this.props
    const { bet } = this.state
    const past = moment(match.dateTime).isBefore(new Date())

    return (
      <Card className="match-card">
        <CardContent className="match-content">
          <div className="match-teams">
            <Bet
              team={teamA}
              betValue={bet.betTeamA}
              onBetValueUpdated={this.handleTeamAChange}
              past={past}
            />
            <Bet
              team={teamB}
              betValue={bet.betTeamB}
              onBetValueUpdated={this.handleTeamBChange}
              past={past}
            />
          </div>
          {!past && <Odds {...match.odds} teamA={teamA} teamB={teamB} />}
          <Scores {...match} />
          <Divider />
          <MatchInfos match={match} />
          {!past && <ValidIcon valid={this.betSaved()} />}
        </CardContent>
      </Card>
    )
  }
}

Match.defaultProps = {
  match: {},
  teamA: {},
  teamB: {},
}

Match.propTypes = {
  match: PropTypes.shape({
    dateTime: PropTypes.instanceOf(Date).isRequired,
    scores: PropTypes.shape({}),
  }),
  teamA: PropTypes.shape({
    code: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }),
  teamB: PropTypes.shape({
    code: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }),
  bet: PropTypes.shape({
    betTeamA: PropTypes.number,
    betTeamB: PropTypes.number,
  }),
  saveBet: PropTypes.func.isRequired,
}

Match.defaultProps = {
  bet: empty,
}

export default Match
