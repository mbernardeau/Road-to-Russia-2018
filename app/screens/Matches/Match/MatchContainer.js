import { lazyload } from 'react-lazyload'
import { connect } from 'react-redux'
import { compose } from 'redux'
import loader from 'hoc-react-loader'

import betReducer, { fetchBet, saveBet } from 'redux/bets'
import teamsReducer, { fetchTeam } from 'redux/teams'
import stadiumsReducer, { fetchStadium } from 'redux/stadiums'

import Match from './Match'

const mapState = (state, { matchId, match: { teamA, teamB, stadium } }) => ({
  bet: betReducer.get(matchId)(state),
  teamA: teamsReducer.get(teamA)(state),
  teamB: teamsReducer.get(teamB)(state),
  stadium: stadiumsReducer.get(stadium)(state),
})

const mapDispatch = (dispatch, { matchId, match: { teamA, teamB, stadium } }) => ({
  load: () => {
    dispatch(fetchBet(matchId))
    dispatch(fetchTeam(teamA))
    dispatch(fetchTeam(teamB))
    dispatch(fetchStadium(stadium))
  },
  saveBet: bet => {
    dispatch(saveBet(matchId, bet))
  },
})

export default compose(
  lazyload({
    height: 135,
    once: true,
    offset: 300,
  }),
  connect(mapState, mapDispatch),
  loader({ print: ['bet', 'teamA', 'teamB', 'stadium'] }),
)(Match)
