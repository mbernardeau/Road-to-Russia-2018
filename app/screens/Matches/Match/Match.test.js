import React from 'react'
import renderer from 'react-test-renderer'
import moment from 'moment'

import Match from './Match'

describe('screens/Match', () => {
  // Mock date to ensure consistancy between tests
  Date.now = jest.fn(() => 1482363367071)

  moment.locale('fr')

  // Mock toLocaleString because of inconsistant behaviour in node (it works correctly in browsers)
  // eslint-disable-next-line
  Number.prototype.toLocaleString = function() {
    return `${this}`
  }

  const props = {
    match: {
      dateTime: 1530205200,
      odds: {
        1: 1.7,
        2: 4.3,
        A: 3.2,
        B: 0,
        N: 3.2,
      },
      phase: '0',
    },
    teamB: {
      code: 'br',
      group: 'G',
      name: 'Brésil',
    },
    teamA: {
      code: 'cl',
      group: 'G',
      name: 'Chili',
    },
    stadium: {
      capacity: 35000,
      city: 'Kaliningrad',
      name: 'Stade de Kaliningrad',
      photo: {
        credit:
          'By A.Savin (Wikimedia Commons · WikiPhotoSpace) (Own work) [FAL], via Wikimedia Commons',
        url:
          'https://upload.wikimedia.org/wikipedia/commons/thumb/2/29/Kaliningrad_05-2017_img72_new_stadium.jpg/512px-Kaliningrad_05-2017_img72_new_stadium.jpg',
      },
    },
    matchId: '-Kj2zgKPFE3FgypNGQ-1',
    userId: 'yWobdSGHFqTYO5yU0s46QYeplen2',
    firebase: {
      ref: {},
      storage: {},
      database: {},
      auth: {},
      set: {},
      setWithMeta: {},
      uniqueSet: {},
      push: {},
      pushWithMeta: {},
      remove: {},
      update: {},
      updateWithMeta: {},
      login: {},
      logout: {},
      updateAuth: {},
      updateEmail: {},
      updateProfile: {},
      uploadFile: {},
      uploadFiles: {},
      deleteFile: {},
      createUser: {},
      resetPassword: {},
      confirmPasswordReset: {},
      verifyPasswordResetCode: {},
      watchEvent: {},
      unWatchEvent: {},
    },
    bet: {
      lastModified: '2017-09-12T18:58:56.825Z',
      teamA: 3,
      teamB: 1,
    },
    saveBet: jest.fn(),
  }

  it('renders correctly', () => {
    const tree = renderer.create(<Match {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
