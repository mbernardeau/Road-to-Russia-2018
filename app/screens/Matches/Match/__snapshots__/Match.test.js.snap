// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/Match renders correctly 1`] = `
<Card
  className="match-card"
>
  <CardContent
    className="match-content"
  >
    <div
      className="match-teams"
    >
      <Bet
        betValue={undefined}
        onBetValueUpdated={[Function]}
        past={true}
        team={
          Object {
            "code": "cl",
            "group": "G",
            "name": "Chili",
          }
        }
      />
      <Bet
        betValue={undefined}
        onBetValueUpdated={[Function]}
        past={true}
        team={
          Object {
            "code": "br",
            "group": "G",
            "name": "BrÃ©sil",
          }
        }
      />
    </div>
    <Divider />
    <Bet
      match={
        Object {
          "dateTime": 1970-01-18T17:03:25.200Z,
          "odds": Object {
            "1": 1.7,
            "2": 4.3,
            "A": 3.2,
            "B": 0,
            "N": 3.2,
          },
          "phase": "0",
        }
      }
    />
    <Bet
      valid={false}
    />
  </CardContent>
</Card>
`;
